src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let res = '';
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (this[i] === null || typeof this[i] === 'undefined') {
+        this[i] = '';
+      }
+
+      if (separator === null) {
+        res += this[i] + 'null';
+      }
+
+      if (typeof separator === 'undefined') {
+        res += this[i] + ',';
+      }
+
+      if (typeof separator !== 'undefined' && separator !== null) {
+        res += this[i] + separator;
+      }
```

Comment:
На этом этапе ты уже должен проверить separator на undefined/null, и тут без ифа плюсовать
```suggestion
      res += this[i] + separator;
```